docs/core/extensions/localization.md

```diff
@@ -0,0 +1,233 @@
+
+The locale at a bare minimum should define the language, but it can also define the culture (dialect), and even the country. These segments are commonly delimited by the `-` character. With the added specificity of a culture, the "culture fallback" rules are applied where best matches are prioritized. The locale should map to a well-known language tag, see <xref:System.Globalization.CultureInfo.Name?displayProperty=nameWithType>.
+
+### Culture fallback scenarios
+
+Imagine that your localized app supports various Serbian locales, and has the following resource files for its `MessageService`:
+
+| File                             | Dialect                       | Country Code |
+|----------------------------------|-------------------------------|--------------|
+| *MessageService.sr-Cyrl-RS.resx* | (Cyrillic, Serbia)            | RS           |
+| *MessageService.sr-Cyrl.resx*    | Cyrillic                      |              |
+| *MessageService.sr-Latn-BA.resx* | (Latin, Bosnia & Herzegovina) | BA           |
+| *MessageService.sr-Latn-ME.resx* | (Latin, Montenegro)           | ME           |
+| *MessageService.sr-Latn-RS.resx* | (Latin, Serbia)               | RS           |
+| *MessageService.sr-Latn.resx*    | Latin                         |              |
+| *MessageService.sr.resx*         | <sup>†</sup> Latin            |              |
+| *MessageService.resx*            |                               |              |
+
+_<sup>†</sup>  The default dialect for the language._
+
+When your app is running with the <xref:System.Globalization.CultureInfo.CurrentCulture?displayProperty=nameWithType> set to a culture of `"sr-Cyrl-RS"` localization attempts to resolve files in the following order:
+
+1. *MessageService.sr-Cyrl-RS.resx*
+1. *MessageService.sr-Cyrl.resx*
+1. *MessageService.sr.resx*
+1. *MessageService.resx*
+
+However, if your app was running with the <xref:System.Globalization.CultureInfo.CurrentCulture?displayProperty=nameWithType> set to a culture of `"sr-Latn-BA"` localization attempts to resolve files in the following order:
+
+1. *MessageService.sr-Latn-BA.resx*
+1. *MessageService.sr-Latn.resx*
+1. *MessageService.sr.resx*
+1. *MessageService.resx*
+
+The "culture fallback" rule will ignore locales when there are no corresponding matches, meaning resource file number four is selected if it's unable to find a match. If the culture was set to `"fr-FR"`, localization would end up falling to the *MessageService.resx* file which can be problematic.
+
+### Resource lookup
+
+Resource files are automatically resolved as part of a lookup routine. If your project file name is different than the root namespace of your project, the assembly name might differ. This can prevent resource lookup from being otherwise successful. To address this mismatch, use the <xref:Microsoft.Extensions.Localization.RootNamespaceAttribute> to provide a hint to the localization services. When provided it is used during resource lookup.
```

Comment:
```suggestion
Resource files are automatically resolved as part of a lookup routine. If your project file name is different than the root namespace of your project, the assembly name might differ. This can prevent resource lookup from being otherwise successful. To address this mismatch, use the <xref:Microsoft.Extensions.Localization.RootNamespaceAttribute> to provide a hint to the localization services. When provided, it is used during resource lookup.
```